languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -8788,6 +8788,31 @@ though, and has since been slightly modified)-->
     </rulegroup>
   </category>
   <category id="CAT_GRAMMAIRE" name="Grammaire">
+      <rulegroup id= "CA_VA_MAL_SE_FINIR" name="ça va mal se finir" type="style" default="temp_off">
+          <rule>
+              <pattern>
+                  <token regexp="yes">ça</token>
+                  <token regexp="yes">va</token>
+                  <token regexp="yes">mal</token>
+                  <marker><token regexp="yes">se</token></marker>
+                  <token regexp="yes">finir</token>
+              </pattern>
+              <message>Dans cette expression française familière,« se finir » est une erreur grammaticale.</message>
+              <suggestion><match no="4" regexp_match="se" regexp_replace=""/></suggestion>
+              <example correction="">Je pense que ça va mal <marker>se</marker> finir.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token regexp="yes">ça</token>
+                  <token regexp="yes">va</token>
+                  <token regexp="yes">mal</token>
+                  <marker><token regexp="yes">finir</token></marker>
+              </pattern>
+              <message>L’ expression française « ça va mal finir » est familière. Souhaitez-vous écrire l'expression « ça va mal se <suggestion><match no="5" regexp_match="finir" regexp_replace="terminer"/></suggestion> »
+                  employée dans le langage courant ?</message>
+              <example correction="terminer">Je pense que ça va mal <marker>finir</marker>.</example>
+          </rule>
```

Comment:
The first rule corrects "Ça va mal se finir" → "Ça va mal finir". The second rule corrects "Ça va mal finir" → "Ça va mal terminer". So a user who writes "Ça va mal se finir" goes through two steps before getting it right.
Could you compactify this? Maybe this can be done in one rule, with a `<token min="0">se</token>` followed by `<token regexp="yes">finir|terminer</token>`. Both those tokens would then be replaced with just "terminer".

Make sense?